testing/integration/upgrade_integrations_server_test.go

```diff
@@ -0,0 +1,127 @@
+	// to produce FIPS-capable artifacts), using each as the start version for the upgrade.
+	// Try upgrading to the latest version.
+	startVersions := getUpgradeStartVersions(t)
+
+	endVersion := define.Version()
+
+	prov, err := ess.NewProvisioner(ess.ProvisionerConfig{
+		Identifier: "it-upgrade-integrations-server",
+		APIKey:     echApiKey,
+		Region:     echRegion,
+	})
+	require.NoError(t, err)
+	prov.SetLogger(t)
+	statefulProv, ok := prov.(*ess.StatefulProvisioner)
+	require.True(t, ok)
+
+	for _, startVersion := range startVersions {
+		t.Run(fmt.Sprintf("%s_to_%s", startVersion.String(), endVersion), func(t *testing.T) {
+			// Create ECH deployment with start version
+			t.Logf("Creating ECH deployment with version [%s] in region [%s]", startVersion.String(), echRegion)
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			deployment, err := statefulProv.Create(ctx, common.StackRequest{
+				ID:      "it-upgrade-integrations-server",
+				Version: startVersion.String(),
+			})
+			require.NoError(t, err)
+			t.Cleanup(func() {
+				if deployment.ID == "" {
+					// Nothing to cleanup
+					return
+				}
+
+				if t.Failed() {
+					cleanupDelay := 1 * time.Minute
+					t.Logf("Cleaning up ECH deployment [%s] in region [%s] after [%s]", deployment.ID, echRegion, cleanupDelay)
+					<-time.After(cleanupDelay)
+				} else {
+					t.Logf("Cleaning up ECH deployment [%s] in region [%s]", deployment.ID, echRegion)
+				}
+
+				err = prov.Delete(context.Background(), deployment)
+				require.NoError(t, err, "failed to delete deployment after test")
+			})
+
+			// Check that deployment is ready and healthy after creation
+			t.Logf("Waiting for ECH deployment [%s] in region [%s] to be ready and healthy after creation", deployment.ID, echRegion)
+			deployment, err = prov.WaitForReady(context.Background(), deployment)
+			require.NoError(t, err)
+
+			// Upgrade deployment to end version
+			t.Logf("Upgrading ECH deployment [%s] in region [%s] from version [%s] to [%s]", deployment.ID, echRegion, startVersion.String(), endVersion)
+			err = prov.Upgrade(context.Background(), deployment, endVersion)
+			require.NoError(t, err)
+			deployment.Version = endVersion
+
+			// Check that deployment is ready and healthy after upgrade
+			t.Logf("Waiting for ECH deployment [%s] in region [%s] to be ready and healthy after upgrade", deployment.ID, echRegion)
+			deployment, err = prov.WaitForReady(context.Background(), deployment)
+			require.NoError(t, err)
+		})
+	}
+}
+
+func getUpgradeStartVersions(t *testing.T) version.SortableParsedVersions {
+	versions, err := upgradetest.GetUpgradableVersions()
+	require.NoError(t, err, "could not get upgradable versions")
+
+	filteredVersions := make([]*version.ParsedSemVer, 0)
+	for _, ver := range versions {
+		if ver.Less(*upgradetest.Version_8_19_0_SNAPSHOT) {
+			continue // Skip versions older than 8.19.0
+		}
```

Comment:
We should also skip all 9.0.x versions as we are not producing a FIPS build for that release

Reply:
Done in 9bd1130e59.